### PR TITLE
Wrap functions and optimize workflow

### PR DESCRIPTION
**Pull request by XGWang0**
Re warp some functions to simplify or optimize work-flow.

Test result refers to http://147.2.212.248/tests/656#live.

### CODE REVIEW

File: lib/proxymode.pm [hunk 1]
```diff
@@ -127,4 +127,73 @@ sub redirect_serial {
     select_console "root-console";
     $self->reset_curr_serialdev();
 }
+
+sub get_consoledev {
+    if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+        script_run("clear");
+        my $prd_version = script_output("cat /etc/SuSE-release");
```
Comment (okurz) [inline lib/proxymode.pm]: just FYI, the new way is `/etc/os-release`. Don't rely on `/etc/SuSE-release` in the future

File: lib/proxymode.pm [hunk 2]
```diff
@@ -127,4 +127,73 @@ sub redirect_serial {
...
+sub generate_grub {
+    my ($self) = @_;
+    my $ipmi_console = get_consoledev();
+    #only support grub2
+    my $grub_default_file = "/etc/default/grub";
+    my $grub_cfg_file     = "/boot/grub2/grub.cfg";
+
+    my $cmd = "if [ -d /boot/grub2 ]; then cp $grub_default_file ${grub_default_file}.org; sed -ri '/GRUB_CMDLINE_(LINUX|LINUX_DEFAULT|XEN_DEFAULT)=/ {s/(console|com\\d+|loglevel|log_lvl|guest_loglvl)=[^ \"]*//g; /LINUX=/s/\"\$/ loglevel=5 console=$ipmi_console,115200 console=tty\"/;/XEN_DEFAULT=/ s/\"\$/ log_lvl=all guest_loglvl=all console=com2,115200\"/;}' $grub_default_file ; fi";
+    script_run("$cmd");
+    wait_idle 3;
```
Comment (okurz) [inline lib/proxymode.pm]: please avoid wait_idle: https://progress.opensuse.org/issues/5830

File: lib/proxymode.pm (hunk 1 above)
Comment (XGWang0) [inline lib/proxymode.pm]: Okay

Comment (coolo) [inline lib/proxymode.pm]: it's deprecated since sle12 ga - and will most likely never go away :)

**Comment by okurz** [comment]
please rebase on current master and make sure your "os-autoinst" subfolder is up to date to use an updated style check. You may have to create a symlink to `.perltidyrc` manually, e.g. with `ln -s os-autoinst/.perltidyrc`

File: lib/proxymode.pm [hunk 3]
```diff
@@ -127,4 +127,75 @@ sub redirect_serial {
     select_console "root-console";
     $self->reset_curr_serialdev();
 }
+
+sub get_consoledev {
```
Comment (kravciak) [inline lib/proxymode.pm]: This seems very simillar to https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L97

Comment (XGWang0) [inline lib/proxymode.pm]: yes, you are right, i wrap this function is used to flexibly calling under this project.

**Pull request merged**